pkg/coremodel/playlist/coremodel.cue

```diff
@@ -0,0 +1,49 @@
+package playlist
+
+import (
+	"github.com/grafana/thema"
+)
+
+thema.#Lineage
+name: "playlist"
+seqs: [
+	{
+		schemas: [
+			{//0.0
+				// Unique numeric identifier for the playlist.
+				// TODO: Isolate or remove identifiers local to a Grafana instance
+				id?: int64 @reviewme()
+				// Unique playlist identifier. Generated on creation, either by the
+				// creator of the playlist of by the application.
+				uid?: *"" | string & len(uid)<=80 @reviewme()
+				// Name of the playlist.
+				name?: *"" | string & len(name)<=255 @reviewme()
+				// Interval
+				// TODO: Figure out the type of this, and validate that it's how long between shifts between items in the playlist.
+				interval?: *"5m" | string & len(interval)<=255 @reviewme()
+				// Playlist item
+				items?: #Items
+
+				///////////////////////////////////////
+				// Definitions (referenced above) are declared below
+
+				#Items: {
```

Comment:
shouldn't this be an array?

Reply:
🤦

But that makes OpenAPI not work, I've asked Sam about that

Reply:
An array should be fine - there are some known issues with the openapi encoder, but the basic/most important case of an open list with at least zero elements of some type (`[...#Items]`) works fine